### DISCUSSION TITLE
Azure LB Creation Issue

### DISCUSSION THREAD

**Discussion by subash-subramaniyan** [strimzi/strimzi-kafka-operator#7474] [Q&A] [answered]
Hi Team,

Unable to create Azure internal LB, Please help me to fix the issue.

apiVersion: kafka.strimzi.io/v1beta2
kind: Kafka
metadata:
  name: prod1
spec:
  kafka:
    template:
      externalBootstrapService:
        metadata:
          annotations:
            service.beta.kubernetes.io/azure-load-balancer-internal: "true"
      perPodService:
        metadata:
          annotations:
            service.beta.kubernetes.io/azure-load-balancer-internal: "true"
    version: 3.2.0
    replicas: 1
    listeners:
      - name: plain
        port: 9092
        type: internal
        type: loadbalancer
        tls: false
      - name: tls
        port: 9093
        type: internal
        tls: true
        configuration:
          bootstrap:
            annotations:
              service.beta.kubernetes.io/azure-load-balancer-internal: "true"
          brokers:
          - broker: 0
            annotations:
              service.beta.kubernetes.io/azure-load-balancer-internal: "true"
    config:
      offsets.topic.replication.factor: 1
      transaction.state.log.replication.factor: 1
      transaction.state.log.min.isr: 1
      default.replication.factor: 1
      min.insync.replicas: 1
      inter.broker.protocol.version: "3.2"
    storage:
      type: jbod
      volumes:
      - id: 0
        type: persistent-claim
        size: 5Gi
        deleteClaim: false
  zookeeper:
    replicas: 1
    storage:
      type: persistent-claim
      size: 5Gi
      deleteClaim: false
  entityOperator:
    topicOperator: {}
    userOperator: {}

**Comment by scholzj** [comment]
Can you please format it properly? Without the proper formatting with the whitespaces etc., it is hard to say what might be wrong.

**Reply by subash-subramaniyan** [reply]
Sure

**Reply by subash-subramaniyan** [reply]
[kafka.zip](https://github.com/strimzi/strimzi-kafka-operator/files/9777668/kafka.zip)

**Reply by subash-subramaniyan** [reply]
Not able to upload formated file, Please review the file and do the needfull ASAP

**Reply by scholzj** [reply]
You have mixed up the listeners I think. You have the annotations configured for the `type: internal` listener. But that needs to be with the `type: loadbalancer` listener. So I think you need to fix that.

If that still doesn't help, share the updated custom resource. But you should also check the services:
* What state are they in
* If they have the right annotation

**Reply by subash-subramaniyan** [reply]
changed the type to loadblancer, But its not working , Please find the attached image

![image](https://user-images.githubusercontent.com/93960578/195633850-a14e32cc-3ce3-4ccd-bf62-741babdc599b.png)

I have deploy the kafka using cluster.yaml file which is ziped in above the file, Please review and do the needfull.

**Reply by scholzj** [reply]
Ok, well ... did you checked the services then as I suggested?

**Comment by subash-subramaniyan** [comment]
[kafka.zip](https://github.com/strimzi/strimzi-kafka-operator/files/9777657/kafka.zip)


Please find the ZIP fle , Im not able to format, Kindly do the needful

**Comment by subash-subramaniyan** [comment, marked as answer]
I will check the services and let you know, Can you please help whiche is
best way to deploy the strimzi Kafka helm or cluster.yaml file

**Reply by scholzj** [reply]
> Can you please help whiche is
best way to deploy the strimzi Kafka helm or cluster.yaml file

That is up to you. I personally don't use Helm for anything.

**Reply by subash-subramaniyan** [reply]
Thank you
